### ISSUE TITLE
D411: Missing blank line before section -- false positive

### ISSUE THREAD

**Issue by pepoluan** [astral-sh/ruff#7216]
The code looks like this:

```python
class JobProgress:
    """
    Tracks progress by generating job batches and recording the last issued job.
    
    Attributes
    ----------
        next_coordinate(int):
    """
```

Ruff <kbd>D411</kbd> complains that a blank line must precede the `Attributes` section.

But as you can see, there *is* actually a blank line ... though it actually contains 4 spaces, this is due to the auto-indent feature of my IDE (PyCharm).

Such a line (containing only spaces) should not trigger <kbd>D411</kbd> because visually they are the same.

I am using **ruff 0.0.277** on Windows.
